src/utils/string_utils.hpp

```diff
@@ -34,14 +36,22 @@ enum class text_alignment { left, right, center };
 /// Trim from start
 static inline std::string& ltrim(std::string& s) {
     s.erase(s.begin(),
-            std::find_if(s.begin(), s.end(), std::not1(std::ptr_fun<int, int>(std::isspace))));
+            std::find_if(s.begin(), s.end(), std::not1(
+                std::function<int(int)>(
```

Comment:
I think using a lambda instead of `std::function`, as it's easier to read, and has less overhead (and I think the cast isn't required?)

Reply:
Couldn't stop myself, had to look up what the overhead is.

Top is w/ std::function, bottom is w/ lambda (note - my impl. might be wrong, I haven't tested it).
https://godbolt.org/z/EpF2tH

I doubt for this application that overhead is that important (although, I suppose you marked it as `inline`, so maybe it is :)), but the difference surprised me.  I'd even guess that b/c the function is so long the inliner would't follow the suggestion.

Reply:
Thanks @mgeplf :)
Indeed the lambda translates much nicer.
Actually I had just stumbled over this code because `std::ptr_fun` is depricated since C++14 and removed in C++17 and since I had changed compiler flags I started getting those ugly warnings, so thought to fix this up. Without thinking too much I replaced it by `std::function` and was astonished that the compiler was not able to resolve the correct `std::isspace` without the nasty static cast. So... either way, I prefer the lambda implementation.

Reply:
Cool.  The nagging feeling I had when I saw `std::function` probably has a direct lineage to `std::ptr_fun`, now that I think about it :)